insta/src/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+
+const withAuthenticate = Posts => Login =>
+  class extends React.Component {
+    state = {
+      loggedIn: false
+    };
+    changeStatus = () => {
+        this.setState(prevState => {
+            return {
+            loggedIn: !prevState.loggedIn
+            }
+        })
+    }
+
+    logOut = () => {
+        this.setState({
+            loggedIn: false
+        })
```

Comment:
I would recommend also clearing `localStorage` here. You could do something like `localStorage.removeItem('user')`.